src/ITensorTDVP.jl

```diff
@@ -1,9 +1,25 @@
 module ITensorTDVP
 
-using Reexport: @reexport
-@reexport using ITensors.ITensorMPS: tdvp, dmrg_x, to_vec, TimeDependentSum, linsolve
-
+using ITensors: ITensors, MPO
+using ITensors.NDTensors: @Algorithm_str
 using ITensors.ITensorMPS: ITensorMPS
-dmrg(args...; kwargs...) = ITensorMPS.alternating_update_dmrg(args...; kwargs...)
+using KrylovKit: KrylovKit
+
+tdvp(args...; kwargs...) = ITensorMPS.itensortdvp_tdvp(args...; kwargs...)
+
+function KrylovKit.linsolve(A::MPO, args...; kwargs...)
+  return ITensorMPS.itensortdvp_linsolve(A, args...; kwargs...)
+end
+
+dmrg(args...; kwargs...) = ITensorMPS.itensortdvp_dmrg(args...; kwargs...)
+
+dmrg_x(args...; kwargs...) = ITensorMPS.itensortdvp_dmrg_x(args...; kwargs...)
+
+function ITensors.contract(alg::Algorithm"fit", args...; kwargs...)
+  return ITensorMPS.itensortdvp_contract(alg, args...; kwargs...)
+end
```

Comment:
I think to be more conservative we should be as particular as possible with these signatures, i.e. constrain the types of the arguments as much as possible in a way that preserves backwards compatibility.

That we, we have more freedom in `ITensors.ITensorMPS` or `ITensorMPS.jl` to potentially overload any of these functions with some other type signature.

Reply:
Good point. I already had to add the `alg::Algorithm"fit"` argument to `contract` because without that it was "gobbling" too many types of arguments and causing an ambiguity error. So that's actually a helpful "red flag" that the other method definitions might do the same unless we contrain them more. So I'll do that.